kong/db/schema/typedefs.lua

```diff
@@ -57,6 +57,24 @@ local function validate_name(name)
 end
 
 
+local function validate_sni(host)
+  local res, err_or_port = utils.normalize_ip(host)
+  if type(err_or_port) == "string" and err_or_port ~= "invalid port number" then
+    return nil, "invalid value: " .. host
+  end
+
+  if err_or_port == "invalid port number" or type(res.port) == "number" then
+    return nil, "must not have a port"
+  end
+
+  if res.type ~= "name" then
+    return nil, "must not be an IP"
+  end
+
+  return true
+end
```

Comment:
If I were to specify `ip:port`, I would first receive an error stating that I should not specify a port, and after retrying, I would get an error that I should also not specify an IP. I think the `"most not be an IP address"` error should be thrown first.

Also, are we lacking tests for this function?

Reply:
> Also, are we lacking tests for this function?

It was being tested indirectly via the upstreams tests (as is the case for some other typedefs).